test/extended/router/gatewayapi.go

```diff
@@ -0,0 +1,168 @@
+				e2e.Logf("Found the Gateway API CRD named: %v", crd.Name)
+			}
+		})
+
+		g.It("should not be deleted", func() {
+			ctx := context.Background()
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Trying to delete the CRDs")
+			for i := range crdNames {
+				err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Delete(ctx, crdNames[i], metav1.DeleteOptions{})
+				o.Expect(err).To(o.HaveOccurred())
+				e2e.Logf("Got error: %v", err.Error())
+				o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+			}
+		})
+
+		g.It("should not be updated", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Get the CRDs firstly, add spec.names.shortNames then update CRD")
+			for i := range crdNames {
+				crd, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), crdNames[i], metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				// some CRDs have a shortName but some not, just trying to add one for all
+				crd.Spec.Names.ShortNames = append(crd.Spec.Names.ShortNames, "fakename")
+				_, err = crdClient.ApiextensionsV1().CustomResourceDefinitions().Update(context.Background(), crd, metav1.UpdateOptions{})
+				o.Expect(err).To(o.HaveOccurred())
+				e2e.Logf("Got error: %v", err.Error())
+				o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+			}
+		})
+
+		g.It("should not be created", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			fakeCRDName := "fakeroutes.gateway.networking.k8s.io"
+			g.By("Creating new CRD in the group")
+			fakeCRD := buildGWAPICRDFromName(fakeCRDName)
+			e2e.Logf("The CRD to be created is: %v", fakeCRD)
+			_, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Create(context.Background(), fakeCRD, metav1.CreateOptions{})
+			o.Expect(err).To(o.HaveOccurred())
+			e2e.Logf("Got error: %v", err.Error())
+			o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+		})
+	})
+
+	g.Describe("The ValidatingAdmissionPolicy for Gateway API", func() {
+		// Adding [Serial] since it could break Gateway API CRDs creation/deletion/update tests
+		g.It("should be recreated after it is deleted [Serial]", func() {
+			var (
+				ctx        = context.Background()
+				policyName = "openshift-ingress-operator-gatewayapi-crd-admission"
+				interval   = 5 * time.Second
+				// VAP should be recreated in 5 minutes
+				timeout = 5 * time.Minute
+			)
+
+			g.By("The VAP should exist in cluster")
+			_, err := oc.AdminKubeClient().AdmissionregistrationV1().ValidatingAdmissionPolicies().Get(ctx, policyName, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("Delete the VAP")
+			err = oc.AdminKubeClient().AdmissionregistrationV1().ValidatingAdmissionPolicies().Delete(ctx, policyName, metav1.DeleteOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("The VAP should be recreated in allowed time")
```

Comment:
I'm not fully convinced that we have to test this part. 1. It's CVO logic, not ingress operator's, 2. It increases the runtime of the origin tests by ~5 minutes (they take a lot of time already).

Reply:
Yes, good question. I also hesitate if we need to add this test. In the original test plan the case is "Deleting the VAP fails", but indeed the VAP is not protected and can be deleted.
Do you think if we have another test for CRDs other than this? maybe @Miciah has some ideas ?

Reply:
> Do you think if we have another test for CRDs other than this? 

The thing I tested it in the pre-submits in CIO repository was the ability to manipulate CRDs when VAP is not there. For this, I scaled down CVO, removed VAP and tried to remove CRDs which should be possible (without VAP). Then I had to scale the CVO back up, btw this recreates the VAP almost immediately.

Reply:
Yes, seems removing VAP/CRDs with scaling down/up CVO should be quicker than this. Will rewrite this. Thanks